syntax = "proto3";
package csce438;
import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service CoordService{
  rpc Heartbeat (ServerInfo) returns (Confirmation) {}
  rpc GetServer (ID) returns (ServerInfo) {}
  rpc GetSyncServers (Empty) returns (AllSyncServers) {}
  rpc RegisterSyncServer(ServerInfo) returns (Confirmation) {}
}

service SynchService {
  rpc GetUserTLFL(ID) returns (AllData) {}
}

message ServerInfo{
  int32 serverID = 1;
  string hostname = 2;
  string port = 3;
  string type = 4;
  string clusterID = 5;
}

message Confirmation{
  bool status = 1;
  string type = 2;
} 

message ID{
  int32 id = 1;
}

message AllSyncServers {
  repeated ServerInfo servers = 1;
}

message Empty {
  
}

message AllData {
  repeated UserTLFL data = 1;
}

message UserTLFL {
    repeated string tl = 1;
    repeated string flw = 2;
    repeated string flr = 3;
    string user = 4;
    repeated bool status = 5;
}
